src/merge.rs

```diff
@@ -468,16 +468,118 @@ impl<'t, D: Driver, A: AbortSignal> Merger<'t, D, A> {
                     )?;
                 }
             }
+
+            ConflictResolution::Unchanged => {
+                // The children are the same, so we only need to merge one side.
+                for (local_child_node, remote_child_node) in
+                    local_node.children().zip(remote_node.children())
+                {
+                    self.signal.err_if_aborted()?;
+                    self.merge_unchanged_child_into_merged_node(
+                        &mut merged_node,
+                        local_node,
+                        local_child_node,
+                        remote_node,
+                        remote_child_node,
+                    )?;
+                }
+            }
+        }
+
+        if local_node.diverged() {
```

Comment:
We don't check `local_node.validity` because `Validity::Reupload` for a local item doesn't make sense.